client/src/components/Explorer/ExplorerPanel.js

```diff
@@ -70,23 +70,22 @@ class ExplorerPanel extends React.Component {
   }
 
   onItemClick(id, e) {
-    const { pushPage } = this.props;
+    const { gotoPage } = this.props;
 
     e.preventDefault();
     e.stopPropagation();
 
-    pushPage(id);
+    gotoPage(id, 1);
   }
 
   onHeaderClick(e) {
-    const { path, popPage } = this.props;
-    const hasBack = path.length > 1;
+    const { page, gotoPage } = this.props;
 
-    if (hasBack) {
+    if (page.parent) {
```

Comment:
Related to my comment in `ExplorerHeader` – this check isn’t equivalent to the previous one, as pages have a defined `parent` even if the current user technically doesn’t have permissions over that parent.

Here, this means clicking the header causes a JS error (trying to load the details of a parent page the user has no access to) instead of opening the link to the user’s top-level page. I imagine we could fix it by having this be based on `depth` rather than `page.parent`?